data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/client/CloudWatchLogsDispatcher.java

```diff
@@ -0,0 +1,145 @@
+import software.amazon.awssdk.services.cloudwatchlogs.model.CloudWatchLogsException;
+import software.amazon.awssdk.services.cloudwatchlogs.model.InputLogEvent;
+import software.amazon.awssdk.services.cloudwatchlogs.model.PutLogEventsRequest;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+
+public class CloudWatchLogsDispatcher implements Runnable {
+    public static final long UPPER_RETRY_TIME_BOUND = 20000;
+    public static final float EXP_TIME_SCALER = 1.5F;
+    private static final Logger LOG = LoggerFactory.getLogger(CloudWatchLogsDispatcher.class);
+    private final BlockingQueue<ThreadTaskEvents> taskQueue;
+    private final CloudWatchLogsClient cloudWatchLogsClient;
+    private final CloudWatchLogsMetrics cloudWatchLogsMetrics;
+    private final String logGroup;
+    private final String logStream;
+    final int retryCount;
+    final long backOffTimeBase;
+    public CloudWatchLogsDispatcher(final BlockingQueue<ThreadTaskEvents> taskQueue,
+                                    final CloudWatchLogsClient cloudWatchLogsClient,
+                                    final CloudWatchLogsMetrics cloudWatchLogsMetrics,
+                                    final String logGroup, final String logStream,
+                                    final int retryCount, final int backOffTimeBase) {
+        this.taskQueue = taskQueue;
+        this.cloudWatchLogsClient = cloudWatchLogsClient;
+        this.cloudWatchLogsMetrics = cloudWatchLogsMetrics;
+        this.logGroup = logGroup;
+        this.logStream = logStream;
+        this.retryCount = retryCount;
+        this.backOffTimeBase = backOffTimeBase;
+    }
+
+    private List<InputLogEvent> prepareInputLogEvents(final ThreadTaskEvents eventData) {
+        List<InputLogEvent> logEventList = new ArrayList<>();
+
+        for (byte[] data: eventData.getEventMessages()) {
+            InputLogEvent tempLogEvent = InputLogEvent.builder()
+                    .message(new String(data))
+                    .timestamp(System.currentTimeMillis())
+                    .build();
+            logEventList.add(tempLogEvent);
+        }
+
+        return logEventList;
+    }
+
+    /**
+     * Flush function to handle the flushing of logs to CloudWatchLogs services;
+     * @param inputLogEvents Collection of inputLogEvents to be flushed
+     * @return true if successful, false otherwise
+     */
+    public boolean dispatchLogs(List<InputLogEvent> inputLogEvents, Collection<EventHandle> eventHandles) {
+        boolean failedPost = true;
+        int failCounter = 0;
+
+        PutLogEventsRequest putLogEventsRequest = PutLogEventsRequest.builder()
+                .logEvents(inputLogEvents)
+                .logGroupName(logGroup)
+                .logStreamName(logStream)
+                .build();
+
+        //TODO: Could also continue to retry even with InterruptedException instead of directly pushing to DLQ.
```

Comment:
Your TODO comments are contradictory. Do you want to keep retrying even after an exception or do you want to push to DLQ?

Reply:
Apologies for the confusion, I have removed that TODO comment as we only want to push to DLQ when we hit an exception or timeout during transmission.